src/target/riscv/riscv-013.c

```diff
@@ -390,18 +390,29 @@ static void dump_field(struct target *target, int idle, const struct scan_field
 	unsigned int in_address = in >> DTM_DMI_ADDRESS_OFFSET;
 
 	log_printf_lf(LOG_LVL_DEBUG,
-			__FILE__, __LINE__, "scan",
+			__FILE__, __LINE__, __PRETTY_FUNCTION__,
 			"%db %s %08x @%02x -> %s %08x @%02x; %di",
 			field->num_bits, op_string[out_op], out_data, out_address,
 			status_string[in_op], in_data, in_address, idle);
 
-	char out_text[decode_dmi(target, NULL, out_address, out_data) + 1];
-	unsigned int out_len = decode_dmi(target, out_text, out_address, out_data);
-	char in_text[decode_dmi(target, NULL, in_address, in_data) + 1];
-	unsigned int in_len = decode_dmi(target, in_text, in_address, in_data);
-	if (in_text[0] || out_text[0]) {
-		log_printf_lf(LOG_LVL_DEBUG, __FILE__, __LINE__, "scan", "%.*s -> %.*s",
-				out_len, out_text, in_len, in_text);
+	const bool decode_out = out_op == DTM_DMI_OP_WRITE;
+	char out_decoded[decode_dmi(target, NULL, out_address, out_data) + 1];
+	const char *out_text = "";
+	if (decode_out)
+		out_text = decode_dmi(target, out_decoded, out_address, out_data)
+			? out_decoded
+			: "<no decoding available>";
```

Comment:
"no decoding available" sounds a bit odd. We could decode it if we wanted to. We (probably) have the relevant field information available to decode. How about "<value ignored>"?

Reply:
Addressed.